### PR TITLE
Profile telemetry

### PR DESCRIPTION
**Pull request by sarangan12**
It is related to https://github.com/Azure/azure-sdk-for-ruby/issues/1120. 

Added code changes to add telemetry to the profiles.

@veronicagg @salameer Please review and approve.

### CODE REVIEW

File: azure_sdk/lib/latest/modules/analysisservices_profile_module.rb [hunk 1]
```diff
@@ -39,12 +39,18 @@ def initialize(configurable, base_url=nil, options=nil)
...
           end
-
+          add_telemetry(client_0)
           @servers = client_0.servers
 
           @model_classes = ModelClasses.new
         end
 
+        def add_telemetry(client)
+          profile_information = 'Profiles/Rollup/Mgmt/AnalysisServices'
```
Comment (veronicagg) [inline azure_sdk/lib/latest/modules/analysisservices_profile_module.rb]: is there a reason for us not to follow same order (similar string) as in the namespace, for example record.. "Azure/Profiles/Latest/AnalysisServices/Mgmt"
Does the full telemetry string end up looking something like this: "Profiles/Rollup/Mgmt/AnalysisServices/Latest/azure_mgmt_analysis_services/0,15,2" ? Are we able to infer the api-versions from this user agent? In the case of the person using a profile, I think we could by checking the configuration of the "Latest" profile, in the case of using a specific api-version, would we be able to detect which api-version is being used? I'm trying to look at this overall, so we know telemetry gives us what we need.

Comment (sarangan12) [reply azure_sdk/lib/latest/modules/analysisservices_profile_module.rb]: Done. I have changed the format. The version is always available from the additional string. So, we do not have to add them here too

File: generators/profilegen/src/resources/templates/rollup/module_template.template [hunk 2]
```diff
@@ -42,6 +43,12 @@ module Azure::Profiles::<%= @profile_name %>
         @model_classes = ModelClasses.new
       end
 
+      def add_telemetry(client)
+        profile_information = 'Profiles/Rollup/<%= @module_name %>'
```
[the remark this answers is not in the document]
Comment (sarangan12) [reply generators/profilegen/src/resources/templates/rollup/module_template.template]: Done

File: azure_sdk/lib/latest/modules/analysisservices_profile_module.rb (hunk 1 above)
Comment (veronicagg) [reply azure_sdk/lib/latest/modules/analysisservices_profile_module.rb]: I believe you're referring to the version of the gem, I was wondering about the api-version and if what we get overall would be enough for what we want. Thoughts? For profiles itself seems enough, though I believe we'd need to look at the profile configuration for that gem version to know which api-version got picked up, so was wondering if we wanted to consider adding something else.

**Pull request merged**